roles/istio/tasks/certificates.yaml

```diff
@@ -0,0 +1,144 @@
+---
+
+################### Start Get Istio External IP ###################
+
+#- name: Wait for External IP Address to be ready
+#  community.kubernetes.k8s_info:
+#    api_version: v1
+#    kind: Service
+#    name: istio-ingressgateway
+#    namespace: istio-system
+#    wait: yes
+#    wait_timeout: 360
+#  register: istio_web_service
+#  until: istio_web_service.resources[0].status.loadBalancer.ingress[0].ip != ""
+#  retries: 5
+#  delay: 2
+
+
+- name: Set fact about Istio Ingress Gateway External IP
+  set_fact:
+    istio_external_ip: "35.123.14.14141" #"{{ istio_web_service.resources[0].status.loadBalancer.ingress[0].ip }}"
+
+################### End Get Istio External IP ###################
+
+- name: Create temporary certificates directory
+  ansible.builtin.tempfile:
+    state: directory
+    suffix: certificate
+  register: tmp_cert_dir
+
+################### Start Create Certificates ###################
+################### Start Create CA Certificate ###################
+
+- name: Generate CA Private Key with RSA 2048
+  community.crypto.openssl_privatekey:
+    path: "{{ tmp_cert_dir.path }}/ca.key"
+    size: 2048
+
+
+- name: Generate CA CSR
+  community.crypto.openssl_csr:
+    path: "{{ tmp_cert_dir.path }}/ca.csr"
+    privatekey_path: "{{ tmp_cert_dir.path }}/ca.key"
+    common_name: "{{ istio_external_ip }}"
+    basic_constraints:
+      - "CA:TRUE"
+    basic_constraints_critical: true
+
+
+- name: Generate CA Self Signed Certificate
+  community.crypto.x509_certificate:
+    path: "{{ tmp_cert_dir.path }}/ca.crt"
+    privatekey_path: "{{ tmp_cert_dir.path }}/ca.key"
+    csr_path: "{{ tmp_cert_dir.path }}/ca.csr"
+    provider: selfsigned
```

Comment:
> This probably needs to be wrapped in a block so that it is optionally allowed, but the user should be able to pass in their own CA. 

Let's open a follow up issue to allow users to provide their own certs and we will pick it up if the need  arise